### PR TITLE
Don't check fingerprint if 'Content' is unknown

### LINKED ISSUE #507: Unable to add lxd trust certificate when certificate content is not known at plan time
I believe I've hit a bug in the logic for the `lxd_trust_certificate` resource. 

When the certificate content is not known at plan time (e.g. when it's derived from a computed value returned from another resource) the provider exits with `Error: failed to parse certificate`. 

For example the following resources:

```terraform
resource "tls_private_key" "client_key" {
  algorithm   = "ECDSA"
  ecdsa_curve = "P384"
}

resource "tls_self_signed_cert" "client_cert" {
  private_key_pem = tls_private_key.client_key.private_key_pem

  subject {
    common_name  = "just-testing"
    organization = "LXD"
  }

  validity_period_hours = 87600

  allowed_uses = [
    "client_auth"
  ]
}

resource "lxd_trust_certificate" "computed_cert_example" {
  name    = "just-testing"
  content = tls_self_signed_cert.client_cert.cert_pem
}
```

Result in the following plan:
```terraform
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform planned the following actions, but then encountered a problem:

  # tls_private_key.client_key will be created
  + resource "tls_private_key" "client_key" {
      + algorithm                     = "ECDSA"
      + ecdsa_curve                   = "P384"
      + id                            = (known after apply)
      + private_key_openssh           = (sensitive value)
      + private_key_pem               = (sensitive value)
      + private_key_pem_pkcs8         = (sensitive value)
      + public_key_fingerprint_md5    = (known after apply)
      + public_key_fingerprint_sha256 = (known after apply)
      + public_key_openssh            = (known after apply)
      + public_key_pem                = (known after apply)
      + rsa_bits                      = 2048
    }

  # tls_self_signed_cert.client_cert will be created
  + resource "tls_self_signed_cert" "client_cert" {
      + allowed_uses          = [
          + "client_auth",
        ]
      + cert_pem              = (known after apply)
      + early_renewal_hours   = 0
      + id                    = (known after apply)
      + is_ca_certificate     = false
      + key_algorithm         = (known after apply)
      + private_key_pem       = (sensitive value)
      + ready_for_renewal     = false
      + set_authority_key_id  = false
      + set_subject_key_id    = false
      + validity_end_time     = (known after apply)
      + validity_period_hours = 87600
      + validity_start_time   = (known after apply)

      + subject {
          + common_name  = "just-testing"
          + organization = "LXD"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.
╷
│ Error: Failed to parse certificate "just-testing"
│
│   with lxd_trust_certificate.computed_cert_example,
│   on main.tf line 21, in resource "lxd_trust_certificate" "computed_cert_example":
│   21: resource "lxd_trust_certificate" "computed_cert_example" {
│
│ Invalid certificate file
```

### PR DESCRIPTION
**Pull request by jsimpso**
Closes #507

### CODE REVIEW

**Comment by MusicDin** [comment]
Great catch. Thanks for your contribution and thorough description of the issue.
We should just add one more test to prevent future regression.

File: internal/truststore/trust_certificate.go [hunk 1]
```diff
@@ -161,6 +161,10 @@ func (r *TrustCertificateResource) ModifyPlan(ctx context.Context, req resource.
 			return
 		}
 	}
+	// Confirm that `certContent` is not an unknown value
```
Comment (MusicDin) [inline internal/truststore/trust_certificate.go:164]: A small nit, but can you also add a new line before the command and a dot at the end of the comment? Just for consistency. Thanks.

Comment (jsimpso) [reply internal/truststore/trust_certificate.go]: No worries, have pushed that update :+1:

**Comment by jsimpso** [comment]
Rebased to include #511

**Review by MusicDin — APPROVED** [review]
Thanks! Would you be interested in adding one more test to prevent potential regression in the future?

**Pull request merged**